bson/src/main/org/bson/BsonBinaryReader.java

```diff
@@ -415,17 +416,18 @@ public void reset() {
     protected class Mark extends AbstractBsonReader.Mark {
         private final int startPosition;
         private final int size;
+        private final BsonInputMark bsonInputMark;
 
         protected Mark() {
             super();
             startPosition = BsonBinaryReader.this.getContext().startPosition;
             size = BsonBinaryReader.this.getContext().size;
-            BsonBinaryReader.this.bsonInput.mark(Integer.MAX_VALUE);
+            bsonInputMark = BsonBinaryReader.this.bsonInput.getMark(Integer.MAX_VALUE);
         }
 
         public void reset() {
             super.reset();
-            BsonBinaryReader.this.bsonInput.reset();
+            bsonInputMark.reset();
```

Comment:
Needs a null check or could NPE.

Reply:
The bsonInputMark field is always set in the constructor.  How could it NPE?

Reply:
Sorry misread.